### PR TITLE
8252389: Fix mistakes in FX API docs

### PR DESCRIPTION
**Pull request by nlisker**
The usual doc fixes (for OpenJFX 16).

Can wait until RDP2 to see if something else comes up.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8252389](https://bugs.openjdk.java.net/browse/JDK-8252389): Fix mistakes in FX API docs

### Reviewers
 * [Ajit Ghaisas](https://openjdk.java.net/census#aghaisas) (@aghaisas - **Reviewer**)
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/380/head:pull/380`
`$ git checkout pull/380`

### CODE REVIEW

**Comment by kleopatra** [comment]
wondering how small a doc error is small enough to go into this: just noticed that Skinnable talks about "Control" in all of its methods - should be "Skinnable" everywhere (except as example implementation in overview doc), IMO. But then, the overview doc is .. lame as well, so might need its own bug anyway?

**Comment by nlisker** [comment]
I don't mind making the "Control" -> "Skinnable" changes if you detail them, but rewriting the class doc should be it's own issue, in which case you can fix these mistakes there as well.

**Comment by kleopatra** [comment]
thanks, that's about the lines of my thoughts as well :) Will file a separate doc bug for skinnable cleanup

**Comment by kleopatra** [comment]
> 
> 
> thanks, that's about the lines of my thoughts as well :) Will file a separate doc bug for skinnable cleanup

fyi: just filed https://bugs.openjdk.java.net/browse/JDK-8260364

**Review by aghaisas — APPROVED** [review]

**Comment by kevinrushforth** [comment]
@nlisker I'll wait to review this until after you add the fix noted in [JDK-8260430](https://bugs.openjdk.java.net/browse/JDK-8260430).

**Comment by nlisker** [comment]
I added a fix for JDK-8260430. Note that the underlying behavior in this class is different than that of `Node`. While both have a map of properties, `Node` uses a special key to add/get the user data to the map while `MenuItem` has a separate field for it. In `Node` these methods were convenience methods, but here they have a separate functionality.

File: modules/javafx.base/src/main/java/javafx/collections/ListChangeListener.java [hunk 1]
```diff
@@ -192,7 +192,7 @@ public boolean wasAdded() {
          * Indicates if elements were removed during this change.
          * Note that using set will also produce a change with {@code wasRemoved()} returning
          * true. See {@link #wasReplaced()}.
-         * @return true if something was removed from the list
+         * @return {@code true} if something was removed from the list
```
Comment (kevinrushforth) [inline modules/javafx.base/src/main/java/javafx/collections/ListChangeListener.java:195]: This same change could be applied to `wasAdded`.

File: modules/javafx.controls/src/main/java/javafx/scene/control/MenuItem.java [hunk 2]
```diff
@@ -503,24 +503,20 @@ public void fire() {
...
 
     /**
-     * Returns a previously set Object property, or null if no such property
-     * has been set using the {@link MenuItem#setUserData(java.lang.Object)} method.
+     * Returns the {@code Object} that was set by {@link #setUserData(Object)}, or {@code null} if no object has been
+     * set.
      *
-     * @return The Object that was previously set, or null if no property
-     *          has been set or if null was set.
+     * @return the user object that was stored (including {@code null}), or {@code null} if no object has been set.
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/MenuItem.java:509]: Trailing `.` can be removed.

**Review by kevinrushforth — COMMENTED** [review]
I left two additional comments. Otherwise good.

**Review by kevinrushforth — APPROVED** [review]

**Review by aghaisas — APPROVED** [review]

**Pull request closed without merging**